homeworks/week6/NgocTin-Le.java

```diff
@@ -0,0 +1,48 @@
+package com.dsa;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ShortestSubstringOfAllCharacters {
+    public static void main(String[] args) {
+        System.out.println(getShortestSubstringOfAllCharacters("xyyzyzxy", "xyz"));  // "zxy"
+    }
+
+    static String getShortestSubstringOfAllCharacters(String arr, String str) {
+
+        // Edge cases
+        if (arr.isEmpty() || str.isEmpty()) {
+            return "";
+        }
+
+        Map<Character, Integer> targetStr = new HashMap<>();
+
+        // Create hash map of target str to find on S
+        for (int i = 0; i < str.length(); i++) {
+            if (!targetStr.containsKey(str.charAt(i))) {
+                targetStr.put(str.charAt(i), 0);
+            } else {
+                int count = targetStr.get(str.charAt(i));
+                targetStr.put(str.charAt(i), ++count);
+            }
+        }
+
+        Map<Character, Integer> findSubSrt = new HashMap<>();
+        int startW = 0;
+        int endW = 0;
```

Comment:
Your overall approach is right, using two moving indices and a map count is correct.

But how do you move the indices and how do you maintain map count may not be correct.